pkg/apis/mysql/v1alpha1/mysqlcluster_types.go

```diff
@@ -158,6 +159,22 @@ type MysqlClusterSpec struct {
 	// InitFileExtraSQL is a list of extra sql commands to append to init_file.
 	// +optional
 	InitFileExtraSQL []string `json:"initFileExtraSQL,omitempty"`
+
+	// ExtraVolumes allows adding extra volumes to the statefulset
+	// +optional
+	ExtraVolumes []core.Volume `json:"extraVolumes,omitempty"`
```

Comment:
Let's just call them `volumes`, `volumeMounts`, `containers` and `initContainers` and put them under `podSpec`. In next version, we can add a `podTemplate` and use it as a starting point for statefullset pod template.